ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetBlockBodyValidator.java

```diff
@@ -106,48 +107,75 @@ public boolean validateBodyLight(
     return true;
   }
 
-  private static boolean validateTransactionsRoot(final Bytes32 expected, final Bytes32 actual) {
+  private static boolean validateTransactionsRoot(
+      final BlockHeader header, final Bytes32 expected, final Bytes32 actual) {
+    LOG.info(
+        "Invalid block {} ({}): transaction root mismatch (expected={}, actual={})",
```

Comment:
(a) log line belongs inside the if block.
(b) Could this be done with just the header?

Reply:
(a) good catch! There was a second one too.
(b) yes, realised I can use header.toLogString() here as well